internal/backend/backend.go

```diff
@@ -28,6 +31,10 @@ type Backend struct {
 
 	// storeBuilder builds stores for the backend users.
 	storeBuilder store.Builder
+
+	/// loginErrorCount login failure counter
+	loginErrorCount int
```

Comment:
Can you use sync atomics rather than a mutex here?

Reply:
done